administrator/components/com_media/models/api.php

```diff
@@ -38,85 +46,126 @@ public function __construct($config = array())
 	{
 		parent::__construct($config);
 
+		if (!isset($config['providers']))
+		{
+			$config['providers'] = Joomla\CMS\Plugin\PluginHelper::getPlugin('filesystem');
+			$this->providers = $config['providers'];
```

Comment:
What happens when config['providers'] is set?
This should be moved outside of the if block to allow setting providers via config.